toc/rfc/rfc-draft-generic-per-route-features.md

```diff
@@ -0,0 +1,127 @@
+# Meta
+[meta]: #meta
+- Name: Generic Per-Route Features
+- Start Date: 2024-01-08
+- Author(s): @maxmoehl, @domdom82
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (fill in with PR link after you submit it)
+
+## Summary
+
+## Problem
+
+Implementing per-route features in Cloud Foundry is a tedious process because it requires
```

Comment:
Another layer to this problem is many of these configurations are on a foundation level only. It doesn't make sense for the foundation to control these things for _every_ app. App devs want to control these properties but they can't. We _could_ expose them 1 by 1, but it takes work (as you say here).

Reply:
There is the option to discard this RFC and take a different route. For example we could spend some time collecting all the features we want to expose and implement them at once to reduce the "fixed" overhead from implementing each feature on its own. But we prefer the flexibility as there will most certainly be new features down the road.